Shared/Supporting Files/Views/SampleRow.swift

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct SampleRow: View {
+    /// The sample displayed in the row.
+    private let sample: Sample
+    
+    /// The text to bold.
+    private let boldedText: String
+    
+    /// A Boolean value that indicates whether to show the sample's description.
+    @State private var isShowingDescription = false
+    
+    /// Creates a sample row.
+    /// - Parameters:
+    ///   - sample: The sample to display.
+    ///   - boldedText: A string to be bolded in the sample's name or description.
+    init(sample: Sample, boldedText: String = "") {
+        self.sample = sample
+        self.boldedText = boldedText
+    }
+    
+    var body: some View {
+        NavigationLink {
+            SampleDetailView(sample: sample)
+                .id(sample.name)
+        } label: {
+            HStack {
+                VStack(alignment: .leading, spacing: 5) {
+                    Text(sample.name.boldingFirstOccurrence(of: boldedText))
+                    
+                    if isShowingDescription {
+                        Text(sample.description.boldingFirstOccurrence(of: boldedText))
+                            .font(.caption)
+                            .foregroundColor(.secondary)
+                            .transition(.move(edge: .top).combined(with: .opacity))
+                    }
+                }
+                Spacer()
+                Button {
```

Comment:
FYI: I just noticed that this button is not visible when the row is selected. Is that a known issue?

Reply:
#170 tried to fix that, but it was not merged due to a navigation bug related to [List with selection](https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/170#pullrequestreview-1388986734).

#226 could have get rid of the info button, but the newly proposed disclosure group UI was rejected by a few reviewers.

Reply:
This will be fixed in another PR.